AudiusProject/audius-protocol
# CON-297 Auth check the requester is the primary of the observed user

### Description

This PR is to auth the flow to clear user state if the `forceResync` flag is enabled to the primary of the user by verifying the wallet using api signatures.

this pr includes:
- passing in a new object called `forceResyncConfig` that has signature data for identity verification
- refactoring `wallets` to `wallet` since we do not support multi wallet syncs yet and removes confusion
- refactors `CNodeToSpIdMapManager` to `ContentNodeInfoManager` since we use Content Node info from chain in verifying identities
- refactoring ^ usage across code base
- adding types / typescript to the signature verification fn in the file `secondarySyncFromPrimaryUtils.ts`
- refactoring `secondarySyncFromPrimary` structurally
- update tests / styling
- add tests for signature verification `creator-node/test/secondarySyncFromPrimaryUtils.test.js`

### Tests

- `creator-node/test/secondarySyncFromPrimaryUtils.test.js` for api signature verification unit tests
- ran current sync tests
- manually triggered `immediate = true/false` and `forceResync = true/false` to observe behavior

manual test steps:
1. create user
2. upload tracks for user to get clock state >0
3. send signed sync request from primary to a user secondary
4. check clock status == -1 after the sync
5. check clock status == primary and other secondary clock state after a period of time (to indicate that a resync happened)
(send message if you want to see the script I ran)

### Monitoring - How will this change be monitored? Are there sufficient logs / alerts?

- can observe current sync flow logs
- see `shouldForceResync` logs
